src/routes/activityReports/saveReport.test.js

```diff
@@ -0,0 +1,614 @@
+import faker from '@faker-js/faker';
+import { GOAL_STATUS, OBJECTIVE_STATUS } from '../../constants';
+import SCOPES from '../../middleware/scopeConstants';
+import {
+  ActivityReport,
+  ActivityReportGoal,
+  ActivityRecipient,
+  ActivityReportObjective,
+  ActivityReportObjectiveTopic,
+  Goal,
+  Grant,
+  NextStep,
+  Objective,
+  Recipient,
+  User,
+  Permission,
+  Topic,
+  sequelize,
+} from '../../models';
+import { createReport, saveReport } from './handlers';
+
+describe('saveReport', () => {
+  const mockResponse = {
+    json: jest.fn(),
+    sendStatus: jest.fn(),
+    status: jest.fn(() => ({
+      end: jest.fn(),
+    })),
+  };
+
+  let user;
+
+  const grantAndRecipientId = faker.datatype.number({ min: 999 });
+
+  let firstGoal;
+  let secondGoal;
+  let firstGrant;
+  let secondGrant;
+  let recipient;
+  let firstTopic;
+  let secondTopic;
+
+  let firstReport;
+  let secondReport;
+
+  beforeAll(async () => {
+    user = await User.create({
+      homeRegionId: 1,
+      hsesUsername: faker.internet.email(),
+      hsesUserId: `fake${faker.unique(() => faker.datatype.number({ min: 1, max: 10000 }))}`,
+      email: faker.internet.email(),
+      phoneNumber: faker.phone.phoneNumber(),
+      name: faker.name.findName(),
+      role: ['Grants Specialist'],
+    });
+
+    await Permission.create({
+      userId: user.id,
+      regionId: 1,
+      scopeId: SCOPES.READ_WRITE_REPORTS,
+    });
+
+    recipient = await Recipient.create({
+      name: faker.company.companyName(),
+      id: grantAndRecipientId,
+    });
+
+    firstGrant = await Grant.create({
+      id: grantAndRecipientId,
+      recipientId: grantAndRecipientId,
+      number: String(grantAndRecipientId),
+      startDate: new Date(),
+      endDate: new Date(),
+      stateCode: 'AL',
+      status: 'Active',
+      regionId: 1,
+    });
+
+    const secondGrantId = faker.datatype.number({ min: 999 });
+
+    secondGrant = await Grant.create({
+      id: secondGrantId,
+      recipientId: grantAndRecipientId,
+      number: String(secondGrantId),
+      startDate: new Date(),
+      endDate: new Date(),
+      stateCode: 'AL',
+      status: 'Active',
+      regionId: 1,
+    });
+
+    firstTopic = await Topic.create({
+      name: 'New topic',
+    });
+
+    secondTopic = await Topic.create({
+      name: 'New topic 2',
+    });
+
+    // GOAK, I find it very funny
+    const goalName = `GOAK ${faker.datatype.number({ min: 999 })}`;
+
+    firstGoal = await Goal.create({
+      name: goalName,
+      createdVia: 'rtr',
+      endDate: new Date(),
+      isRttapa: 'Yes',
+      grantId: grantAndRecipientId,
+      status: GOAL_STATUS.DRAFT,
+    });
+
+    secondGoal = await Goal.create({
+      name: goalName,
+      createdVia: 'rtr',
+      endDate: new Date(),
+      isRttapa: 'Yes',
+      grantId: secondGrantId,
+      status: GOAL_STATUS.DRAFT,
+    });
+  });
+
+  afterAll(async () => {
```

Comment:
independentHooks: true needs to be on all updates and destroy to make sure any hooks get run appropriately